### PR TITLE
Added support for Model.getModel({ through : { where : { ... } }})

### PR DESCRIPTION
**Pull request by mcarboni-redant**
I've make the code on the answer to work :

http://stackoverflow.com/questions/33351537/use-a-column-in-the-jointable-on-the-where/33362317#33362317

### CODE REVIEW

File: lib/associations/belongs-to-many.js [hunk 1]
```diff
@@ -445,6 +445,12 @@ BelongsToMany.prototype.injectGetter = function(obj) {
       throughWhere = {};
       throughWhere[association.foreignKey] = instance.get(association.source.primaryKeyAttribute);
 
+      if (options.through && options.through.where)
```
Comment (janmeier) [inline lib/associations/belongs-to-many.js]: We should probably move this after  `.scope` so through can overwrite the scope if needed

**Comment by janmeier** [comment]
Code LGTM, just a small comment.

We also need an integration test

**Comment by mcarboni-redant** [comment]
OK, moved after the scope and writed a little comment, for the integration test at the moment i can't write one, i can pass you the test i'm using in my project

var widgetSchema = {
        name: {
            type: DataTypes.STRING
        }
    };

var Widget = sequelize.define('widget', widgetSchema);

var userSchema = {
        name: {
            type: DataTypes.STRING
        }
    };

var User = sequelize.define('user', userSchema);

var User_widget = sequelize.define('perms', {
        read: {
            type: DataTypes.BOOLEAN,
            defaultValue: false
        },
        write: {
            type: DataTypes.BOOLEAN,
            defaultValue: false
        }
    });

Widget.belongsToMany(User, { through:User_widget, foreignKey: 'widget_id' });
User.belongsToMany(Widget, { through:User_widget, foreignKey: 'user_id' });

User.create({ id: 1, name : 'test' }).then(function(user) {
   //Make the widgets 
   //..
   widget1.perms = { read : true, write : false };
   widget2.perms = { read : false, write : false };
   widget3.perms = { read : true, write : true };
   user.getWidgets({
       through : {
            where : {
                'read' : true
            }
        }
   }).then(function(widgets) {
     expect(widgets.length).equal.to(2);
  });

**Comment by janmeier** [comment]
That looks fine for a test - just add it somewhere appropriate in https://github.com/sequelize/sequelize/blob/master/test/integration/associations/belongs-to-many.test.js

File: lib/associations/belongs-to-many.js [hunk 2]
```diff
@@ -449,6 +449,13 @@ BelongsToMany.prototype.injectGetter = function(obj) {
         _.assign(throughWhere, through.scope);
       }
 
+      //If a user pass a where on the options through options, make an "and" with the current throughWhere
+      if (options.through && options.through.where)
+        throughWhere = { $and : [
+            throughWhere,
+            options.through.where
+        ]};
+
```
Comment (mickhansen) [inline lib/associations/belongs-to-many.js]: Please add brackets around the if statement.
Please place `$and` on its own line in the object.

No spacing after `$and` before `:`.

**Comment by mcarboni-redant** [comment]
I can't setup the tests to work in my environment, so i can't be sure the tests i push works fine

**Comment by luin** [comment]
+1 for this feature

**Comment by felixfbecker** [comment]
Closing in favor of #5479

**Pull request closed without merging**